### PR TITLE
1.15/1438 reinstate a family after leaving vv eee

### PR DESCRIPTION
**Pull request by lauragurney**
This is huge and likely to clash with @charlesstrange2 big PR too but thought I'd get this up so we can work on sorting the merge conflicts that will inevitably occur. Plus it needs sense checking as I've been looking at it for too long. Tests to follow later.

https://trello.com/c/ZDSD9FsP/1438-reinstate-a-family-after-leaving-vv-eee

### CODE REVIEW

File: database/migrations/2023_03_15_161618_add_fields_to_families.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('families', function (Blueprint $table) {
+            $table->timestamp('rejoin_on')->nullable();
+            $table->integer('leave_amount')->default(0);
```
Comment (neonleann) [inline database/migrations/2023_03_15_161618_add_fields_to_families.php:18]: Does this need to be added to the Model? ($fillable)

File: app/Http/Controllers/Store/FamilyController.php [hunk 2]
```diff
@@ -30,10 +33,73 @@ public function update(StoreUpdateRegistrationFamilyRequest $request, Registrati
...
+        // get Family
+        /** @var  $family Family */
+        $family = $registration->family;
+
+        // Set rejoin date
+        $family->rejoin_on = Carbon::now();
+
+        // Wipe previous leaving_on to make them 'current' again - keep original leaving_reason for
+        // reports and to identify them as a previous leaver
+       // $family->leaving_on = null;
```
[the remark this answers is not in the document]
Comment (lauragurney) [reply app/Http/Controllers/Store/FamilyController.php:67]: No I need to remove that

File: resources/views/store/view_registration.blade.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+@section('title', 'Check / Update Registration')
+
+@section('content')
+
+@include('store.partials.navbar', ['headerTitle' => 'Check or update'])
+
+<div class="content flex">
+    <form action="{{ URL::route("store.registration.rejoin",['registration' => $registration]) }}" method="post" class="full-height">
+        {{ method_field('PUT') }}
+        {!! csrf_field() !!}
```
[the remark this answers is not in the document]
Comment (charlesstrange2) [reply resources/views/store/view_registration.blade.php:12]: this was the technique from before `@csrf` existed. prolly best to switcn to `@csrf` though

File: resources/views/store/partials/other_info.blade.php [hunk 4]
```diff
@@ -128,8 +128,8 @@
                 <i class="fa fa-ticket button-icon" aria-hidden="true"></i>Go to voucher manager
             </div>
         </a>
-
-        @if (!isset($registration->family->leaving_on) )
+        @php(\App\Http\Controllers\Store\FamilyController::status($registration))
```
[the remark this answers is not in the document]
Comment (lauragurney) [reply resources/views/store/partials/other_info.blade.php:131]: Urgh, I need to remove the if statement

File: database/migrations/2023_03_15_161618_add_fields_to_families.php (hunk 1 above)
Comment (lauragurney) [reply database/migrations/2023_03_15_161618_add_fields_to_families.php:18]: Done

**Review by neonleann — APPROVED** [review]
👪 This PR has been approved!

**Review by charlesstrange2 — CHANGES_REQUESTED** [review]
You can probably remove the new formrequest?

File: app/Http/Requests/StoreRejoinRegistrationFamilyRequest.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        // A reason key might be present.
+        return [
+            'leaving_reason' => [
```
[the remark this answers is not in the document]
Comment (lauragurney) [reply app/Http/Requests/StoreRejoinRegistrationFamilyRequest.php]: I've amended the rules to an empty array, removing the form fails validation so it is needed.

**Review by charlesstrange2 — APPROVED** [review]
Okedoke

**Pull request merged**